### PR TITLE
copyright: add missing license

### PR DESCRIPTION
**Pull request by amergnat**
Solderpad Hardware License

Signed-off-by: Alexandre Mergnat <amergnat@baylibre.com>

### CODE REVIEW

File: soc/riscv/riscv-ite/common/encoding.h [hunk 1]
```diff
@@ -1,8 +1,12 @@
 /*
- * Copyright (c) 2020 ITE Corporation. All Rights Reserved.
- *
- * SPDX-License-Identifier: Apache-2.0
- *
+ * Copyright and related rights are licensed under the Solderpad Hardware
```
Comment (galak) [inline soc/riscv/riscv-ite/common/encoding.h:2]: Is it clear this file needs its license changed?

Comment (ceolin) [reply soc/riscv/riscv-ite/common/encoding.h:2]: yep, AFAIU it is fine keep the license, the case is just add the original copyright

Comment (amergnat) [reply soc/riscv/riscv-ite/common/encoding.h:2]: The original is a modified Apache 2, then I prefer to just put Solderpad Hardware.

**Comment by amergnat** [comment]
@ceolin @galak If there are no other comment/issue, can you please merge it ?

**Comment by ceolin** [comment]
> @ceolin @galak If there are no other comment/issue, can you please merge it ?

I see your point. I really don't know how to proceed with a different license even if it is compatible with Apache 2. @kestewart  do you know how to proceed ?

**Comment by kestewart** [comment]
Has Solderpad gone through license review (sorry if I've missed it)?   I think it should be ok, but it should be reviewed and process followed. 

Can  you please use that text with the SPDX license identifier,  rather than putting the text of the license in the files, so it gets recognized by the tooling.   There is an SPDX identifier for this license, so best to use that rather than the cut/paste text which may be cut/paste error prone.  See: the https://spdx.org/licenses/SHL-0.51.html

**Comment by amergnat** [comment]
@ceolin @galak @kestewart 
I've fixed my patch thanks to kestewart advices (I hope I understood). Do you agree ?

**Comment by ceolin** [comment]
I may be wrong, but my understand is that the original copyright is missing.

**Comment by amergnat** [comment]
> I may be wrong, but my understand is that the original copyright is missing.

I have created this file then I put the original copyright function to where I've found the source. You notify me the source come from Solderpad Hardware, then yes I replaced instead of adding because my source ("the original copyright") was wrong, logic IMO.

>SOLDERPAD HARDWARE LICENSE version 0.51
>This license is based closely on the Apache License Version 2.0, but is not approved or endorsed by the Apache Foundation. A >copy of the non-modified Apache License 2.0 can be found at http://www.apache.org/licenses/LICENSE-2.0.
>As this license is not currently OSI or FSF approved, the Licensor permits any Work licensed under this License, at the option of >the Licensee, to be treated as licensed under the Apache License Version 2.0 (which is so approved).

If you really want keeping Apache-2.0, tell me clearly, because isn't logic for me. But no problem if you ask me to add it I will do it without pleasure.

**Comment by amergnat** [comment]
> I may be wrong, but my understand is that the original copyright is missing.

done

**Comment by amergnat** [comment]
@kgugala @ceolin @galak @kestewart 
Have you other comments ?
If not, could you put your approval for merge this please ?

**Comment by pgielda** [comment]
Wait, but why this change is happening in the first place? Any rationale?

Solderpad is meant for hardware in my opinion...

**Comment by amergnat** [comment]
> Wait, but why this change is happening in the first place? Any rationale?
> 
> Solderpad is meant for hardware in my opinion...

These changes are requested by @ceolin, I let him answer.

**Comment by ceolin** [comment]
> > Wait, but why this change is happening in the first place? Any rationale?
> > Solderpad is meant for hardware in my opinion...
> 
> These changes are requested by @ceolin, I let him answer.

Ok, a lot of back and forth here. What I said since the beginning is that this file was copied from https://github.com/openhwgroup/cva6/blob/master/bootrom/encoding.h and we should keep the right copyright. I honestly don't remember have asked anything about the license. This file license seems to be compatible with Zephyr's license, but I'm not a lawyer and don't want to make any claim about it.

What I always said was that we should keep the copyright of the original author, it is fine to add your copyright for the thing that you have added. But unless someone say something different, these files have copied contents from that project and we should give them the credit.

**Comment by pgielda** [comment]
OK, but if the file is copied, then why it has a different copyright now? Michael Schaffner (the original file committer in the cva6 repo) is from Google (and the file is a header, has no copyright in it, just a mention of a LICENSE file) and the Zephyr file suddenly has "Copyright (c) 2020 BayLibre" ? Am I missing something?

**Comment by ceolin** [comment]
> OK, but if the file is copied, then why it has a different copyright now? Michael Schaffner (the original file committer in the cva6 repo) is from Google (and the file is a header, has no copyright in it, just a mention of a LICENSE file) and the Zephyr file suddenly has "Copyright (c) 2020 BayLibre" ? Am I missing something?

That is my point the whole time. The copyright is just not right IMHO.

**Comment by amergnat** [comment]
> > OK, but if the file is copied, then why it has a different copyright now? Michael Schaffner (the original file committer in the cva6 repo) is from Google (and the file is a header, has no copyright in it, just a mention of a LICENSE file) and the Zephyr file suddenly has "Copyright (c) 2020 BayLibre" ? Am I missing something?
> 
> That is my point the whole time. The copyright is just not right IMHO.

@ceolin  @pgielda The file isn't just copied. there are some modifications / additional content.

**Comment by ceolin** [comment]
my understanding is that you should have both copyrights, the original + yours. e.g:
```
* Copyright (c) 2020 Michael Schaffner
* Copyright (c) 2020 BayLibre
```

**Comment by amergnat** [comment]
@ceolin  Sorry for the delay.
It should be ok now :)

**Comment by amergnat** [comment]
Hello @ceolin @kestewart @pgielda @galak 
Any update ? Hope I understood your comment and my patch is correct now.

**Review by cfriedt — APPROVED** [review]
I'm going to approve this purely because Michael Schaffner was the original author, and that the original LICENSE was SHL.

Whether SHL is compatible with Apache-2.0, and more importantly whether to preserve the original license, is something that should have been considered when the file was first added to the Zephyr tree.

It would be a lesser evil to include the original copyright holder and license than not to, imho.

**Comment by cfriedt** [comment]
@ceolin @kestewart @pgielda @galak - please either nak or ack. Thanks

**Comment by amergnat** [comment]
@ceolin @kestewart @pgielda @galak any update ?

**Review by ceolin — APPROVED** [review]

**Pull request merged**